frame-metadata/src/lib.rs

```diff
@@ -298,63 +375,91 @@ impl core::fmt::Debug for DefaultByteGetter {
 
 /// Hasher used by storage maps
 #[derive(Clone, PartialEq, Eq, Encode)]
-#[cfg_attr(feature = "std", derive(Decode, Serialize, Debug))]
+#[cfg_attr(feature = "std", derive(Decode, Serialize, Deserialize, Debug))]
 pub enum StorageHasher {
```

Comment:
Should this have `#[non_exhaustive]` perhaps? (And maybe elsewhere as well?)

Reply:
Well, since we have metadata versioning anyway, it's totally fine to assume that `V12` has these concrete types.